ts/RootContainer.tsx

```diff
@@ -34,13 +65,52 @@ const styles = StyleSheet.create({
  */
 class RootContainer extends React.Component<Props> {
   public componentDidMount() {
-    AppState.addEventListener("change", this.onApplicationActivityChange);
+    if (Platform.OS === "android") {
+      Linking.getInitialURL()
+        .then(this.navigate)
+        .catch(console.error);
+    } else {
+      Linking.addEventListener("url", this.handleOpenURL);
+    }
+
+    AppState.addEventListener("change", this.handleApplicationActivityChange);
   }
 
   public componentWillUnmount() {
-    AppState.removeEventListener("change", this.onApplicationActivityChange);
+    Linking.removeEventListener("url", this.handleOpenURL);
+    AppState.removeEventListener(
+      "change",
+      this.handleApplicationActivityChange
+    );
   }
 
+  private handleOpenURL = (event: { url: string }) => {
+    this.navigate(event.url);
+  };
+
+  private navigate = (url: string | null) => {
```

Comment:
instead of accepting a `null` here and then returning, accept only `string` and avoid the call at the caller (e.g. line 88):

```ts
if(event.url) {
  this.navigate(event.url);
} 
```

Reply:
The function still have to accept a null argument, it's executed by `Linking.getInitialURL().then()` as well, and the argument could be `null`.